lib/Twig/Template.php

```diff
@@ -607,15 +618,33 @@ protected function getAttribute($object, $item, array $arguments = array(), $typ
             $this->env->getExtension('Twig_Extension_Sandbox')->checkMethodAllowed($object, $method);
         }
 
-        // Some objects throw exceptions when they have __call, and the method we try
-        // to call is not supported. If ignoreStrictCheck is true, we should return null.
-        try {
-            $ret = call_user_func_array(array($object, $method), $arguments);
-        } catch (BadMethodCallException $e) {
-            if ($call && ($ignoreStrictCheck || !$this->env->isStrictVariables())) {
-                return;
+        if ($call) {
+            // Some objects throw exceptions when they have __call, and the method we try
+            // to call is not supported. If ignoreStrictCheck is true, we should return null.
+            try {
+                $ret = $object->__call($method, $arguments);
+            } catch (BadMethodCallException $e) {
+                if ($ignoreStrictCheck || !$this->env->isStrictVariables()) {
+                    return;
+                }
+                throw $e;
             }
-            throw $e;
+        } elseif ($a = $arguments) {
+            switch (false) {
+                case isset($a[1]): $ret = $object->$method($a[0]); break;
+                case isset($a[2]): $ret = $object->$method($a[0], $a[1]); break;
```

Comment:
the old code allows to have associative arrays (even though keys are ignored), because ``call_user_func_array`` ignores keys (except in HHVM 3.14 but this looks like a bug as older HHVM version behave like PHP): https://3v4l.org/M3h42
So this cannot be accepted as it breaks BC (or you need to do ``$a = array_values($arguments)``, but this might actually negate the micro-optimization you are adding)